test/new-e2e/pkg/environments/aws/kubernetes/eks.go

```diff
@@ -324,34 +324,24 @@ func EKSRunFunc(ctx *pulumi.Context, env *environments.Kubernetes, params *Provi
 			if err := fakeIntake.Export(awsEnv.Ctx(), &env.FakeIntake.FakeintakeOutput); err != nil {
 				return err
 			}
+		} else {
+			env.FakeIntake = nil
 		}
 
 		// Deploy the agent
 		dependsOnSetup := utils.PulumiDependsOn(workloadDeps...)
 		if params.agentOptions != nil {
-			paramsAgent, err := kubernetesagentparams.NewParams(params.agentOptions...)
+			params.agentOptions = append(params.agentOptions, kubernetesagentparams.WithPulumiResourceOptions(dependsOnSetup))
+			kubernetesAgent, err := helm.NewKubernetesAgent(&awsEnv, "eks", eksKubeProvider, params.agentOptions...)
 			if err != nil {
 				return err
 			}
-
-			helmComponent, err := agent.NewHelmInstallation(&awsEnv, agent.HelmInstallationArgs{
-				KubeProvider:  eksKubeProvider,
-				Namespace:     "datadog",
-				ValuesYAML:    paramsAgent.HelmValues,
-				Fakeintake:    fakeIntake,
-				DeployWindows: params.eksWindowsNodeGroup,
-			}, dependsOnSetup)
+			err = kubernetesAgent.Export(ctx, &env.Agent.KubernetesAgentOutput)
 			if err != nil {
 				return err
 			}
+		} else {
 			env.Agent = nil
-
-			ctx.Export("agent-linux-helm-install-name", helmComponent.LinuxHelmReleaseName)
-			ctx.Export("agent-linux-helm-install-status", helmComponent.LinuxHelmReleaseStatus)
-			if params.eksWindowsNodeGroup {
-				ctx.Export("agent-windows-helm-install-name", helmComponent.WindowsHelmReleaseName)
-				ctx.Export("agent-windows-helm-install-status", helmComponent.WindowsHelmReleaseStatus)
```

Comment:
The provisioner is not used for now. We should avoid exporting fields like that since we want to use component and leverage code we have to automatically import and match fields with struct